src/machine/machine_nrf_sd.go

```diff
@@ -0,0 +1,56 @@
+//go:build nrf && softdevice
+
+package machine
+
+import (
+	"device/arm"
+	"device/nrf"
+
+	"errors"
+)
+
+// avoid a heap allocation in GetRNG.
+var (
+	softdeviceEnabled uint8
+	bytesAvailable    uint8
+	buf               [4]uint8
+)
+
+// GetRNG returns 32 bits of non-deterministic random data based on internal thermal noise.
+// According to Nordic's documentation, the random output is suitable for cryptographic purposes.
+func GetRNG() (ret uint32, err error) {
+	// First check whether the SoftDevice is enabled.
+	// sd_rand_application_bytes_available_get cannot be called when the SoftDevice is not enabled.
+	arm.SVCall1(0x12, &softdeviceEnabled) // sd_softdevice_is_enabled
+
+	if softdeviceEnabled == 0 {
+		return getRNG()
+	}
+
+	// call into the SoftDevice to get random data bytes available
+	switch nrf.Device {
+	case "nrf51":
+		// sd_rand_application_bytes_available_get: SOC_SVC_BASE_NOT_AVAILABLE + 4
+		arm.SVCall1(0x2B+4, &bytesAvailable)
+	case "nrf52", "nrf52840", "nrf52833":
```

Comment:
Is there a case when we have `"nrf52840", "nrf52833"` and not `"nrf52"`?
I mean this can probably be dropped: `", nrf52840", "nrf52833"`?

Reply:
I stole this code from https://github.com/tinygo-org/tinygo/blob/release/src/runtime/runtime_nrf_softdevice.go#L32